src/portal-impl.c

```diff
@@ -387,61 +387,114 @@ load_portal_configuration_for_dir (gboolean    opt_verbose,
+  if (portal_dir != NULL)
+    {
+      load_config_directory (portal_dir, desktops, opt_verbose);
+      /* All other config directories are ignored when this is set */
+      return;
+    }
+
+  /* $XDG_CONFIG_HOME/xdg-desktop-portal/(DESKTOP-)portals.conf */
+  user_portal_dir = g_build_filename (g_get_user_config_dir (), XDP_SUBDIR, NULL);
 
-  conf = load_portal_configuration_for_dir (opt_verbose, user_portal_dir, "portals.conf");
-  if (conf != NULL)
+  if (load_config_directory (user_portal_dir, desktops, opt_verbose))
+    return;
+
+  /* $XDG_CONFIG_DIRS/xdg-desktop-portal/(DESKTOP-)portals.conf */
+  dirs = g_get_system_config_dirs ();
+
+  for (iter = dirs; iter != NULL && *iter != NULL; iter++)
     {
-      if (opt_verbose)
-        g_debug ("Using user portal configuration file");
+      g_autofree char *dir = g_build_filename (*iter, XDP_SUBDIR, NULL);
 
-      config = g_steal_pointer (&conf);
+      if (load_config_directory (dir, desktops, opt_verbose))
+        return;
     }
 
-  desktops = get_current_lowercase_desktops ();
-  for (size_t i = 0; desktops[i] != NULL; i++)
-    {
-      g_autofree char *portals_conf = g_strdup_printf ("%s-portals.conf", desktops[i]);
+  /* ${sysconfdir}/xdg-desktop-portal/(DESKTOP-)portals.conf */
+  if (load_config_directory (SYSCONFDIR "/" XDP_SUBDIR, desktops, opt_verbose))
+    return;
 
-      conf = load_portal_configuration_for_dir (opt_verbose, user_portal_dir, portals_conf);
-      if (conf != NULL)
-        {
-          if (opt_verbose)
-            g_debug ("Using user portal configuration file '%s' for desktop '%s'",
-                     portals_conf,
-                     desktops[i]);
+  /* $XDG_DATA_HOME/xdg-desktop-portal/(DESKTOP-)portals.conf
+   * (just for consistency with other XDG specifications) */
+  g_clear_pointer (&user_portal_dir, g_free);
+  user_portal_dir = g_build_filename (g_get_user_data_dir (), XDP_SUBDIR, NULL);
 
-          config = g_steal_pointer (&conf);
-          return;
-        }
+  if (load_config_directory (user_portal_dir, desktops, opt_verbose))
+    return;
 
-      conf = load_portal_configuration_for_dir (opt_verbose, portal_dir, portals_conf);
-      if (conf != NULL)
-        {
-          if (opt_verbose)
-            g_debug ("Using system portal configuration file '%s' for desktop '%s'",
-                     portals_conf,
-                     desktops[i]);
+  /* $XDG_DATA_DIRS/xdg-desktop-portal/(DESKTOP-)portals.conf */
+  dirs = g_get_system_data_dirs ();
 
-          config = g_steal_pointer (&conf);
-          return;
-        }
+  for (iter = dirs; iter != NULL && *iter != NULL; iter++)
+    {
+      g_autofree char *dir = g_build_filename (*iter, XDP_SUBDIR, NULL);
+
+      if (load_config_directory (dir, desktops, opt_verbose))
+        return;
     }
+
+  /* ${datadir}/xdg-desktop-portal/(DESKTOP-)portals.conf */
+  if (load_config_directory (DATADIR "/" XDP_SUBDIR, desktops, opt_verbose))
+    return;
```

Comment:
In practice this is redundant with `XDG_DATA_DIRS` but I thought we should probably search it anyway?